cpp/tests/binaryop/assert-binops.h

```diff
@@ -34,8 +34,13 @@ template <typename TypeOut,
           typename TypeLhs,
           typename TypeRhs,
           typename TypeOp,
-          typename ScalarType = cudf::experimental::scalar_type_t<TypeLhs>>
-void ASSERT_BINOP(column_view const& out, scalar const& lhs, column_view const& rhs, TypeOp&& op)
+          typename ValueComparator = std::equal_to<TypeOut>,
+          typename ScalarType      = cudf::experimental::scalar_type_t<TypeLhs>>
+void ASSERT_BINOP(column_view const& out,
+                  scalar const& lhs,
+                  column_view const& rhs,
+                  TypeOp&& op,
+                  ValueComparator const& value_comparator = ValueComparator())
 {
```

Comment:
These changes will probably conflict with https://github.com/rapidsai/cudf/pull/4854  but that's ok.  This method is probably better.